src/simplify_algebra.cpp

```diff
@@ -72,14 +74,37 @@ struct find_mul_conv
         auto a_ins    = r.instructions["a"];
         auto w_ins    = r.instructions["w"];
 
-        auto broadcast_op = any_cast<op::broadcast>(a_ins->get_operator());
-        if(broadcast_op.axis != 1)
+        auto a_input_lens = a_ins->inputs().front()->get_shape().lens();
+
+        std::size_t num_not_one_dims = std::count_if(
+            a_input_lens.cbegin(), a_input_lens.cend(), [](auto dim) { return dim != 1; });
+        if(num_not_one_dims > 1)
+            return;
+
+        // check broadcasted along channels
+        auto a_lens    = a_ins->get_shape().lens();
+        auto a_strides = a_ins->get_shape().strides();
+        // handle len = 1 case
+        auto invalid_sl = [&](std::size_t i) {
+            if(a_strides[i] != 0)
+            {
+                return a_lens[i] != 1;
+            }
+            return false;
+        };
+        auto invalid_case = false;
+        for(int i = 2; i < a_lens.size(); ++i)
+        {
+            if(invalid_sl(i))
+                invalid_case = true;
+        }
```

Comment:
use `std::any_of` ?

Reply:
I don't see a good way to use `std::any_of` as it needs to check `a_lens` and `a_strides`.